src/Domain/ValueObjects/WalletValueObject.cs

```diff
@@ -0,0 +1,19 @@
+using Domain.Aggregates;
+namespace Domain.ValueObjects;
+
+public record Wallet(decimal Balance){
+    public User User { get; set; }
+    private decimal Balance { get; set; } = Balance;
+    
+    public bool HasBalance(decimal amount)
+    {
+        if (amount == Balance) return true;
```

Comment:
what if the balance is more than the amount - item's price?

Reply:
just fxed it